src/api/app/views/webui/request/show.html.erb

```diff
@@ -193,23 +193,14 @@
                   <% if sourcediff[:error] %>
                       <%= content_tag(:p, content_tag(:i, sourcediff[:error], class: :error)) %>
                   <% else %>
-                      <% if action[:type] == :delete
-                           t = sourcediff['old']
-                         else
-                           t = sourcediff['new']
-                         end
-                         project, package, rev = t['project'], t['package'], t['rev'] %>
-
+                      <% diff = diff_data(action[:type], sourcediff) %>
                       <% if action[:sourcediff].length != 1 %>
                           <h4>
-                            <%= sourcediff['new']['project'] %> / <%= sourcediff['new']['package'] %>
-                            (rev <%= sourcediff['new']['rev'] %>) -
-                            <%= sourcediff['old']['project'] %> / <%= sourcediff['old']['package'] %>
-                            (rev <%= sourcediff['old']['rev'] %>)
+                            <%= diff_label(sourcediff['new']) %> - <%= diff_label(sourcediff['old']) %>
                           </h4>
                       <% end %>
                       <%#TODO: Redo source/target link display %>
-                      <%= render(:partial => 'shared/sourcediff', locals: { filenames: sourcediff['filenames'], files: sourcediff['files'], :source => { project: project, package: package, :rev => rev }, :css_id => "diff_action_#{action_index}_#{action[:type]}_#{sourcediff_index}", :editor_width => '550px' }) %>
+                      <%= render(partial: 'shared/sourcediff', locals: { filenames: sourcediff['filenames'], files: sourcediff['files'], source: diff, css_id: "diff_action_#{action_index}_#{action[:type]}_#{sourcediff_index}", editor_width: '550px' }) %>
```

Comment:
Now source can be undefined, before it was always defined, but had `nil` values. This can crash in the partial :fire:

Reply:
There is always a project and a package when there is a sourcediff. So this should be fine.

Reply:
I don't understand what you mean... :confused:

Reply:
I read it wrong, sorry